server/data/openlibrary/check_cover_existance.js

```diff
@@ -2,17 +2,12 @@ const __ = require('config').universalPath
 const requests_ = __.require('lib', 'requests')
 const error_ = __.require('lib', 'error/error')
 
-module.exports = url => {
-  return requests_.head(url)
-  .then(checkHeader.bind(null, url))
-}
-
-const checkHeader = (url, res) => {
-  const contentType = res.headers['content-type']
-  // Coupled with OpenLibrary response headers
-  if (contentType && contentType === 'image/jpeg') {
-    return url
-  } else {
+module.exports = async (url, expectedContentType) => {
+  const { headers } = await requests_.head(url)
+  const contentType = headers['content-type']
+  if (expectedContentType != null && contentType !== expectedContentType) {
```

Comment:
might be more readable to actually [pass a flag `?default=false`](https://openlibrary.org/dev/docs/api/covers) on the request to OL. and expect a 404 when no image available

Reply:
done in 79681beb